src/components/MovieDetails.tsx

```diff
@@ -1,9 +1,120 @@
-const MovieDetails = () => {
+import { useState, useEffect, ReactNode } from 'react';
+
+import api from '../service/api';
+
+import ReadMore from './ReadMore';
+
+import '../styles/MovieDetails.scss';
+
+enum ImageWidth {
+  w500
+}
+
+type MovieDetailsProps = {
+  match: { params: { id: string } }
+}
+
+type MovieInfo = {
+  poster_path: string;
+  original_title: string;
+  title: string;
+  tagline: string;
+  status: string;
+  release_date: string;
+  budget: string;
+  popularity: string;
+  vote_average: string;
+  runtime: string;
+  genres: [{ id: number | null, name: string }],
+  production_countries: [{ name: string }]
+  overview: string;
+};
+
+const MovieDetails = ({ match }: MovieDetailsProps) => {
+  const [movieData, setMovieData] = useState<MovieInfo>({
+    poster_path: '',
+    original_title: '',
+    title: '',
+    tagline: '',
+    status: '',
+    release_date: '',
+    budget: '',
+    popularity: '',
+    vote_average: '',
+    runtime: '',
+    genres: [{ id: null, name: '' }],
+    production_countries: [{ name: '' }],
```

Comment:
Just a note - not sure why we need to have { id: null, name: '' } as initial data, not just an empty array?

Reply:
if its an empty array, an errors appears from the TypeScript: "type [] is not assignable to type that specified in type MovieInfo "